firecracker.go

```diff
@@ -183,16 +183,16 @@ func (f *Client) PutGuestDriveByID(ctx context.Context, driveID string, drive *m
 	return f.client.Operations.PutGuestDriveByID(params)
 }
 
-// PutGuestVsockOpt is a functional option to be used for the
+// PutGuestVsockByIDOpt is a functional option to be used for the
 // PutGuestVsockByID API in setting any additional optional fields.
 type PutGuestVsockByIDOpt func(*ops.PutGuestVsockByIDParams)
 
-// PutGuestVsock is a wrapper for the swagger generated client to make
+// PutGuestVsockByID is a wrapper for the swagger generated client to make
 // calling of the API easier.
 func (f *Client) PutGuestVsockByID(ctx context.Context, vsock *models.Vsock, opts ...PutGuestVsockByIDOpt) (*ops.PutGuestVsockByIDNoContent, error) {
 	params := ops.NewPutGuestVsockByIDParams()
 	params.SetContext(ctx)
-	params.SetID(*vsock.VsockID)
+	params.ID = *vsock.VsockID
```

Comment:
Gah, didn't realize that `ID` was a `string` and not a `*string`. Can you please change this to be what I suggested eariler?
```go
params.SetID(StringValue(vsock.VsockID))
```